nikolai4D/ContechOS
# C2 133 filter view create context menu that changes depending on clicking canvas nodes and rels that are in different levels

## Description & motivation



In this ticket, the context menu was added. This was done by
- Adding props, propKeys, typeDataPropKeys, instanceDataPropKeys to nodes
- Adding props, propKeys typeDataRelPropKeys, instanceDataRelPropKeys to rels
- Adjusting definitions.json (adding filter view, modified parameters for typeData so that the Create Form is correct, modified parameters for configDef for adding Type to context menu)
- Adjustments in the Create Form for typeData to generate typeDataPropKeys directly (instead of firstly choosing parent from drop down)

## How to test



Go to filter view
1. Right click on canvas -> should show "Definition" 
  1.1 Click on Definition -> a create from should show
  
 2. In order to see the different context menu's, we need to click on the nodes/rels on different levels.
- configDef 
- configDef with propKeys
- configObj
- configObj w inherited propKeys
- configObj w typeDataPropKeys and instanceDataPropKeys
- typeData 
- typeData w inherited propKeys
- instanceData w inherited propKeys
 --
- configDefInternalRel/ExternalRel
- configDefInternalRel/ExternalRel with propKeys
- configObjInternalRel/ExternalRel
- configObjInternalRel/ExternalRel w inherited propKeys
- configObjInternalRel/ExternalRel w typeDataPropKeys and instanceDataPropKeys
- typeDataInternalRel/ExternalRel 
- typeDataInternalRel/ExternalRel w inherited propKeys
- instanceDataInternalRel/ExternalRel w inherited propKeys

However, this can be very time consuming. To do this in a less time consuming way, 
- go to the filter box
- tick as many nodes as possible (in a database with a lot of nodes)
- Left click/right click on as many nodes/rels as possible. We should be able to pick anything from the context menu. 
- Open the edit form, by clicking on a node and pressing on the pen in the read form.
 
 
 Disclaimer:
- This ticket is only for viewing context menu. However, the functionality on some of the features in the context menu work already (creating some nodes/rels, deleting nodes/rels, edit nodes). Full CRUD functionality is separated from this ticket.


## New tickets to be added



None

---
